### PR TITLE
Refactor code and add more tests

### PR DESCRIPTION
**Pull request by thiagonache**
Makes the code a bit more readable and maintainable.

### CODE REVIEW

File: ratchet/.gitignore [hunk 1]
```diff
@@ -1,4 +1,9 @@
+# Files generated by Makefile
 schemata
 output/*
 build/*
+
+# Mac meta files
```
Comment (jpluscplusm) [inline ratchet/.gitignore]: 100% fine by me, but I find this usually goes in a local, machine-wide gitignore file!

File: ratchet/ratchet.go [hunk 2]
```diff
@@ -18,100 +17,110 @@ func ValidTFSchemaVersion(schemaVersion gjson.Result) bool {
 	}
 }
 
+func formatPrimitiveTypes(key, value string, typeAttributes gjson.Result) string {
+	var output string
+	switch {
+	case typeAttributes.Get("required").Bool():
+		output = fmt.Sprintf("    %s!: %s", key, value)
```
Comment (jpluscplusm) [inline ratchet/ratchet.go:24]: If it makes the handling of these string-constructing-string any easier (not only here, but everywhere), remember that we can execute a post-ratchet `cue fmt` to fix up all the spacing. In other words, you /could/ stop worrying about space prefixes in this tool's output ...

**Review by jpluscplusm — DISMISSED** [review]

File: ratchet/.gitignore (hunk 1 above)
Comment (thiagonache) [reply ratchet/.gitignore]: done!

File: ratchet/ratchet.go (hunk 2 above)
Comment (thiagonache) [reply ratchet/ratchet.go:24]: for now, it's not a problem.

**Pull request merged**